src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -779,86 +616,21 @@ private void ScanSyntaxToken(ref TokenInfo info)
 
                 case '$':
                     if (TryScanInterpolatedString(ref info))
-                    {
                         break;
-                    }
 
                     if (this.ModeIs(LexerMode.DebuggerSyntax))
-                    {
-                        goto case 'a';
-                    }
+                        goto case '_';
 
                     goto default;
 
-                // All the 'common' identifier characters are represented directly in
-                // these switch cases for optimal perf.  Calling IsIdentifierChar() functions is relatively
-                // expensive.
-                case 'a':
-                case 'b':
-                case 'c':
-                case 'd':
-                case 'e':
-                case 'f':
-                case 'g':
-                case 'h':
-                case 'i':
-                case 'j':
-                case 'k':
-                case 'l':
-                case 'm':
-                case 'n':
-                case 'o':
-                case 'p':
-                case 'q':
-                case 'r':
-                case 's':
-                case 't':
-                case 'u':
-                case 'v':
-                case 'w':
-                case 'x':
-                case 'y':
-                case 'z':
-                case 'A':
-                case 'B':
-                case 'C':
-                case 'D':
-                case 'E':
-                case 'F':
-                case 'G':
-                case 'H':
-                case 'I':
-                case 'J':
-                case 'K':
-                case 'L':
-                case 'M':
-                case 'N':
-                case 'O':
-                case 'P':
-                case 'Q':
-                case 'R':
-                case 'S':
-                case 'T':
-                case 'U':
-                case 'V':
-                case 'W':
-                case 'X':
-                case 'Y':
-                case 'Z':
+                // All the 'common' identifier characters are represented directly in these switch cases for optimal
+                // perf.  Calling IsIdentifierChar() functions is relatively expensive.
+                case (>= 'a' and <= 'z') or (>= 'A' and <= 'Z'):
```

Comment:
I would like to see a microbenchmark on the difference here. These emit different code, and this is perhaps one of the hottest parts of the parsing loop.

Reply:
i should have mentioned.  i did benchmark parsing files, and got no change at all.  i can try an actual benchmark, but it's definitely harder to setup since i'd need the before/after, and direct hooks into it.

Reply:
> i can try an actual benchmark, but it's definitely harder to setup since i'd need the before/after, and direct hooks into it.

I think a small repro project with a bit of copy/pasting from the old and new should be easier to set up.